openid/src/main/java/fish/payara/security/openid/controller/ProviderMetadataContoller.java

```diff
@@ -77,37 +76,47 @@
      *
      */
     public JsonObject getDocument(String providerURI) {
-        if (isNull(providerDocuments.get(providerURI))) {
-            if (providerURI.endsWith("/")) {
-                providerURI = providerURI.substring(0, providerURI.length() - 1);
+        if (!providerDocuments.containsKey(providerURI)) {
+            JsonObject responseObject;
+            if ("".equals(providerURI)) {
+                // no providerURI provided, data must be load from @OpenIdProviderMetadata
+                responseObject = JsonObject.EMPTY_JSON_OBJECT;
+            } else {
+                responseObject = downloadWellKnownUris(providerURI);
             }
+            providerDocuments.put(providerURI, responseObject);
+        }
+        return providerDocuments.get(providerURI);
+    }
 
-            if (!providerURI.endsWith(WELL_KNOWN_PREFIX)) {
-                providerURI = providerURI + WELL_KNOWN_PREFIX;
-            }
+    private JsonObject downloadWellKnownUris(String providerURI) {
+        if (providerURI.endsWith("/")) {
+            providerURI = providerURI.substring(0, providerURI.length() - 1);
+        }
 
-            Client client = ClientBuilder.newClient();
-            WebTarget target = client.target(providerURI);
-            Response response = target.request()
-                    .accept(APPLICATION_JSON)
-                    .get();
+        if (!providerURI.endsWith(WELL_KNOWN_PREFIX)) {
```

Comment:
Shouldn't the constant be named rather `WELL_KNOWN_SUFFIX` if it should be at the end?